test/functional/aws_test.go

```diff
@@ -0,0 +1,162 @@
+)
+
+//TODO update when Fuse Online is ready as per https://gitlab.cee.redhat.com/integreatly-qe/integreatly-test-cases/blob/master/tests/high-availability/f01-verify-all-aws-resources-are-created-in-aws.md
+const (
+	RDSSuffix                  = "postgres-integreatly"
+	UPSResourceName            = "ups-" + RDSSuffix
+	CRResourceName             = "codeready-" + RDSSuffix
+	ClusterSSOResourceName     = "rhsso-" + RDSSuffix
+	ApplicationSSOResourceName = "rhssouser-" + RDSSuffix
+	ThreeScaleResourceName     = "threescale" + RDSSuffix
+	ThreeScaleNameElement      = "rhmioperatorthre"
+	S3Backup                   = "rhmioperatorback"
+	resourceStatusAvailable    = "available"
+	awsCredsNamespace          = "kube-system"
+	awsCredsSecretName         = "aws-creds"
+)
+
+var (
+	expectedResources = []string{
+		ThreeScaleResourceName,
+		UPSResourceName,
+		CRResourceName,
+		ClusterSSOResourceName,
+		ApplicationSSOResourceName,
+	}
+)
+
+func TestAWS(t *testing.T) {
+	//TODO
+}
+
+func testExpectedResourcesExist(t testing.T, f *framework.Framework, instances []*rds.DBInstance, cacheClusters []*elasticache.CacheCluster, buckets []*s3.Bucket, expectedResources []string) (map[string]bool, error) {
+	err := setAWSCredEnvar(f)
+	if err != nil {
+		fmt.Errorf("error :%w", err)
+		return nil, err
+	}
+	var checkedResources map[string]bool
+	//check RDS resources
+
+	for _, instance := range instances {
+		for _, name := range expectedResources {
+			if instance.DBName != aws.String(name) {
+				continue
+			}
+			if instance.MultiAZ != aws.Bool(true) {
+				t.Logf("instance: %v , has multi-AZ disabled, instance ARN: %v ", instance.DBName, instance.DBInstanceArn)
+				return nil, errors.New("error: Multi-AZ is disabled")
+			}
+			if instance.StorageEncrypted != aws.Bool(true) {
+				t.Logf("instance: %v , has database encryption disabled,  instance ARN: %v ", instance.DBName, instance.DBInstanceArn)
+				return nil, errors.New("error: encryption is disabled")
+			}
+			if instance.PreferredBackupWindow == nil {
+				t.Logf("instance: %v , has automated backups disabled,  instance ARN: %v ", instance.DBName, instance.DBInstanceArn)
+				return nil, errors.New("error: automated backups are disabled")
+			}
+			if instance.DeletionProtection != aws.Bool(true) {
+				t.Logf("instance: %v , has deletion protection disabled,  instance ARN: %v ", instance.DBName, instance.DBInstanceArn)
+				return nil, errors.New("error: deletion protection is disabled")
+			}
+			checkedResources[aws.StringValue(instance.DBName)] = true
+
+		}
+	}
+}
+
+func newAWSClients(session *session.Session) *awsClients {
+	return &awsClients{
+		rdsClient:         rds.New(session),
+		elasticacheClient: elasticache.New(session),
+		s3Client:          s3.New(session),
+	}
+}
+
+func setAWSCredEnvar(f *framework.Framework) error {
+	secret := &corev1.Secret{}
+	err := f.Client.Get(goctx.TODO(), types.NamespacedName{Name: awsCredsSecretName, Namespace: awsCredsNamespace}, secret)
```

Comment:
Why are we using the `kube-system` secret. Should we be using the secret created by CRO?

Reply:
Yes, noticed this yesterday, going to change this in the next commit